src/main/java/messenger/ImageMessengerRequestProcessor.java

```diff
@@ -0,0 +1,55 @@
+package messenger;
+
+import notification.Notification;
+import request.RequestProcessor;
+
+import org.springframework.web.multipart.MultipartFile;
+import org.springframework.web.multipart.MultipartHttpServletRequest;
+
+import javax.servlet.http.HttpServletRequest;
+import java.io.IOException;
+
+/**
+ * Created by annterina on 11.11.17.
+ */
+public class ImageMessengerRequestProcessor extends RequestProcessor {
+
+    MultipartFile image;
+    String imageName;
+    long imageSize;
+
+    @Override
+    protected void preprocess(HttpServletRequest request)
+    {
+        super.preprocess(request);
+        MultipartHttpServletRequest multipartRequest = (MultipartHttpServletRequest) request;
+        image = multipartRequest.getFile("image");
+        imageName = image.getOriginalFilename();
+        imageSize = image.getSize();
+    }
+
+    @Override
+    protected void postprocess(HttpServletRequest request, Notification notification)
+    {
+        super.postprocess(request, notification);
+    }
+
+    @Override
+    protected Notification constructNotification(HttpServletRequest request) {
+
+        try {
+            return new ImageMessengerNotificationBuilder()
+                    .withSenderName(request.getParameter("image-sender-name"))
+                    .withReceiverName(request.getParameter("image-receiver-name"))
+                    .withSendImage(image.getBytes())
+                    .withImageName(imageName)
+                    .withImageSize(imageSize)
+                    .build();
+        } catch (IOException e) {
```

Comment:
This is super weird. Picture yourself as a user of our library. What would you expect to happen, when you want to send image notification and your app does not find an image? I'd say that I either want my app to crash(equivalent of throwing exception), redirect him or at least sending some default image which I'll make sure that will ALWAYS be. (third option is the worst option and should not be considered at all). 

Right now you're gonna send an object of correct type, looking cool and stuff, but it will actually have no image! It will have `null` instead. Now, as a user you see a method declaring "returning ImageNotification" so you expect to have this image. So you probably will access image and boom - NullPointerException.

Not to make this any longer then it should be, let's go with the first option. There is class UncheckedIOException which basically stands for runtime IOException which means "well IO Exception happened, what are we gonna do about it, huh?"
```
try {
            return new ImageMessengerNotificationBuilder()
                    .withSenderName(request.getParameter("image-sender-name"))
                    .withReceiverName(request.getParameter("image-receiver-name"))
                    .withSendImage(image.getBytes())
                    .withImageName(imageName)
                    .withImageSize(imageSize)
                    .build();
        } catch (IOException e) {
             throw UncheckedIOException("Tell sth about image", e);
  }
```

Reply:
It was my clever way to investigate some other issues in my task, but yep, I understand what you mean :) I am changing that, right this second!